lib/elements/rebar_hook.rb

```diff
@@ -0,0 +1,48 @@
+require './db/base'
+require './lib/materials/steel'
+
+module Elements
+  class RebarHook < Materials::Steel
+    MAX_FY = 550.to_f
+    MIN180_HOOK_LENGTH = 60.to_f
+    MIN_HOOK_LENGTH = 75.to_f
+
+    attr_reader :angle, :length, :diameter
+
+    def initialize(yield_stress:, elastic_module:)
+      @diameter = nil
+      @length = nil
+      @angle = nil
+
+      super(
+        yield_stress:,
+        elastic_module:
+      )
+    end
+
+    def setup_properties(diameter, angle)
+      @diameter = diameter
+      use_angle_of(angle)
+    end
```

Comment:
how is this different from the constructor? Why do we need a separate method that sets up properties?

Reply:
Because of the diameter, take a look at the upgraded proposal